### PR TITLE
Support YOLOX and Adapt to PyTorch 1.7.0

### PR DESCRIPTION
**Pull request by wjj19950828**
# Create A Good Pull Request

> 下面的文字请保留在PR说明的最后面，并在提完PR后，根据实际情况勾选确认以下情况  

Please check the follow step before merging this pull request

- [x] Python code style verification
- [x] Review all the code diff by yourself
- [x] All models(TensorFLow/Caffe/ONNX/PyTorch) testing passed
- [x] Details about your pull request, releated issues

If this PR add new model support, please update `model_zoo.md` and add model to out test model zoos(@wjj19950828)
- [x] New Model Supported
- [ ] No New Model Supported

## Do follow contributes
1、add aten::silu op for PyTorch
2、fixd aten::upsample_nearest2d bug
3、Support YOLOX PyTorch Model
4、Remove the to_tensor operation of the code optimization part
5、Add aten::format op，adapt to PyTorch 1.7.0
6、Add enable_code_optim switch

## others
1、repo link: https://github.com/Megvii-BaseDetection/YOLOX
2、commit id: f778fdba8f083e6f1e8dd168fe7a8faa5faa8484

### CODE REVIEW

File: x2paddle/op_mapper/pytorch2paddle/aten.py [hunk 1]
```diff
@@ -4716,6 +4716,70 @@ def aten_sigmoid(mapper, graph, node):
...
+
+    graph.add_layer(
+        "paddle.nn.Silu",
+        inputs=layer_inputs,
+        outputs=layer_outputs,
+        scope_name=scope_name)
+    return current_inputs, current_outputs
+
+
+def aten_silu_(mapper, graph, node):
```
Comment (jiangjiajun) [inline x2paddle/op_mapper/pytorch2paddle/aten.py]: 这里两个函数代码看起来类似，是否可以合并起来

**Review by jiangjiajun — CHANGES_REQUESTED** [review]

File: x2paddle/op_mapper/pytorch2paddle/aten.py (hunk 1 above)
Comment (wjj19950828) [reply x2paddle/op_mapper/pytorch2paddle/aten.py]: TorchScript中会出现这种情况：aten::relu和aten::relu_，这种名字不同但是功能一致的op，之前的逻辑是在添加一个convert函数，复用aten_relu的代码，现在添加判断逻辑，aten::relu和aten::relu_共用aten_relu()这个function

File: x2paddle/convert.py [hunk 2]
```diff
@@ -220,6 +225,7 @@ def pytorch2paddle(module,
                    save_dir,
                    jit_type="trace",
                    input_examples=None,
+                   code_optimizer=True,
```
Comment (jiangjiajun) [inline x2paddle/convert.py]: 改下，命名改成动+名， 这里改成 enable_code_optim

File: x2paddle/core/program.py [hunk 3]
```diff
@@ -237,11 +237,11 @@ def update(layers):
 
         return update(self.layers)
 
-    def gen_model(self, save_dir, jit_type=None):
+    def gen_model(self, save_dir, jit_type=None, code_optimizer=True):
```
Comment (jiangjiajun) [inline x2paddle/core/program.py]: 同上

**Review by jiangjiajun — CHANGES_REQUESTED** [review]

File: x2paddle/convert.py (hunk 2 above)
Comment (wjj19950828) [reply x2paddle/convert.py]: Done.

File: x2paddle/core/program.py (hunk 3 above)
Comment (wjj19950828) [reply x2paddle/core/program.py]: Done.

**Pull request merged**